fix #428: ignore timezone when saving event

jlwt90 opened the pull request:
Hi, this is the PR for fixing #428 .

**event creation**
This has been fixed by adding timezone to date related values in `clean` method in `EventWizardBasicsForm`.

**event update**
I still have no idea how I can fix it. (Sorry I'm pretty new to django..)
>From the top of my head, I would go neither for option 1 nor for option 2. EventWizard and EventUpdate are views, which are rather to high-level for this kind of thing and Event is a model which is rather to low-level for this. In my opinion, the golden middle road would be the forms involved, which are EventWizardBasicsForm and EventUpdateForm. We could do it in the save() methods there but I think an even better way might be the clean() method in Django's validation cycle.

@raphaelm I just learnt the django's validation cycle you mentioned in the comment from #428. Thanks! 👍 👍  However, I am wondering whether we can use the same technique in event update.
Since `timezone` is stored in `EventSettingsForm` while `date_to` (as well as the other 3 fields) are stored in `EventUpdateForm`, I can't use (more like I don't know how to use... ) the timezone to update all 4 fields in `clean` method. Since `EventUpdate` view is the bridge between the two forms, should I also change part of the logic there?

raphaelm commented:
> Since timezone is stored in EventSettingsForm while date_to (as well as the other 3 fields) are stored in EventUpdateForm, I can't use (more like I don't know how to use... ) the timezone to update all 4 fields in clean method. Since EventUpdate view is the bridge between the two forms, should I also change part of the logic there?

Your observation is very correct, I missed that fact. Yes, I think the view is a good place then.

jlwt90 commented:
I updated both `EventUpdate` view and `EventWizardBasicsForm` and added another test case for checking timezone & date time.
However, I couldn't find any test case for event creation (please tell me if there is any). I think I'll create another test case with custom timezone in `EventWizardBasicsForm` later.

raphaelm commented on src/tests/control/test_events.py:
```diff
@@ -52,6 +52,23 @@ def test_settings(self):
+        doc.select("[name=settings-timezone]")[0]['value'] = "Asia/Tokyo"
+        doc.find('option', {"value": "Asia/Tokyo"})['selected'] = 'selected'
+        doc.find('option', {"value": "UTC"}).attrs.pop('selected')
+        print(extract_form_fields(doc.select('.container-fluid form')[0]))
```
Can you remove the ``print`` statement here?

raphaelm commented on src/tests/control/test_events.py:
```diff
@@ -52,6 +52,23 @@ def test_settings(self):
+        # date_to should not be changed even though the timezone is changed
+        assert doc.select("[name=date_to]")[0]['value'] == "2013-12-30 17:00:00"
+        assert doc.find('option', {"value": "Asia/Tokyo"})['selected'] == "selected"
+        assert doc.select("[name=settings-max_items_per_order]")[0]['value'] == "12"
```
Could you not only check what is shown in the form after saving but additionaly also perform a `self.event1.refresh_from_db()` and then check the time that is saved in the database? Just to be double-sure nothing gets lost in the form layer.

jlwt90 replied:
nice idea! thanks!

raphaelm commented:
Nice! This is good work! :clap: Thank you very much for your contribution!

If you want to continue working on pretix, I think you are ready for something beyond the "easy" scope. The issues #308, #372, #297 or #309 come to my mind here. They are of different size but all have a well-defined goal and I can provide you with help on how to do it. Is there one among them that would interest you?

jlwt90 commented:
@raphaelm thanks! I'll certainly continue working on pretix (it's an interesting project!).
I went through the issues you mentioned above and I think I can help on #308, if you don't mind working with newbie.

Since I still have another task at hand, I think I'll probably start working on it 2~3 days later.
Thank you again for your feedback and suggestions! 👏

raphaelm commented:
Sure, take your time and just ask if anything is unclear on #308 :)
